pkg/signature/kms/hashivault/client.go

```diff
@@ -309,7 +323,7 @@ func (h hashivaultClient) verify(sig, digest []byte, alg crypto.Hash, opts ...si
 
 	isValid, ok := valid.(bool)
 	if !ok {
-		return fmt.Errorf("data type assertion for field `valid` failed: %T %#v", valid.(bool), valid.(bool))
+		return fmt.Errorf("received non-bool value from 'valid' key")
```

Comment:
should we use something like %T to still get the type or %s to attempt to stringify it?

Reply:
Reproduced the same case on playground, and it threw panic for the `valid.(bool)` casting: https://go.dev/play/p/-rCmjQsK0e8

But works well if we just use

```go
fmt.Printf("%T %#v", valid, valid)
```

I can change all type assert errors to get their types `%T` and stringify `%#v` but I'm not sure if they are safe to print the entire content to stdout in vault client. 🤔